### ISSUE TITLE
base_dir is not respected for include path using MSVC

### ISSUE THREAD

**Issue by srohmen** [ccache/ccache#1265]
### How to reproduce ###

1. Having a config file which contains:
```
debug = true
base_dir = D:/workspace/test
```
2. Executing `ccache cl /c main.cpp /ID:\workspace\test`
3. Inspecting *.ccache-input-text files

### Actual behavior ###

The log file contains:
```
### arg
/ID:\workspace\test
```

### Expected behavior ###

Before 3781f2567d92f11a41fe5ba8099f8c9a6b5497d2 the output is:
```
### arg
-I.
```
The base_dir is respected nicely for the include path.

### Environment ###

ccache version 4.8

Windows 10, MSVC (VS2019)

**Comment by srohmen** [comment]
Am I right with my observation (and by the commit message), that some options are not known to ccache and thus not converted from /X to -X?

I have similar issues for these options, which suddenly show up with a slash (ccache-4.8) instead of a dash (ccache-4.7) in the log:
```
/FdD:\path\to\checkout\build\zlib\Release\zlib.pdb
/errorReport:prompt
/d1trimfile:D:\path\to\checkout\
```

**Comment by jrosdahl** [comment]
Thanks for the bug report. A fix for this (if it does not break #1230) would be welcome.

> Am I right with my observation (and by the commit message), that some options are not known to ccache

Yes, ccache does not maintain knowledge of all compiler options. It strives to know about options that matter, though.

> and thus not converted from /X to -X?

No, all `/` options are converted to `-` options internally: https://github.com/ccache/ccache/blob/1519c9aa1d89608f2690584a15ff5d4d021b27ed/src/argprocessing.cpp#L272-L276

However, an intentional side effect of 3781f2567d92f11a41fe5ba8099f8c9a6b5497d2 is that `/` options now are passed as `/` options to the compiler -- before the internal `-` representation was passed to the compiler.

The commit message of 3781f2567d92f11a41fe5ba8099f8c9a6b5497d2 talks about whether an argument that isn't a known option is to be considered an input file. For instance, in `cl.exe /c /path/to/file.c` the `/c` option is known to be an option (and therefore not an input file). Before 3781f2567d92f11a41fe5ba8099f8c9a6b5497d2, `/path/to/file.c` was considered an option since it starts with `/` and ccache therefore would not find the source file in the command line. After 3781f2567d92f11a41fe5ba8099f8c9a6b5497d2 ccache figures out that it is a source file if the file exists.

> I have similar issues for these options, which suddenly show up with a slash (ccache-4.8) instead of a dash (ccache-4.7) in the log:

Is it a problem that `/` options are not converted to `-`?

**Comment by jrosdahl** [comment]
The `/I...` part of the problem was fixed in ac05df21de6a72d50614ee64c6e4752846625593 (part of ccache v4.10). However, 6a5c4de998af28a4b00c1c31a368ec7bf8412b0b (also part of ccache 4.10) introduced a bug where it no longer works if `base_dir` ends with a slash.
